src/InfluxDB.LineProtocol/LineProtocolWriter.cs

```diff
@@ -0,0 +1,239 @@
+﻿using System;
+using System.Globalization;
+using System.IO;
+
+namespace InfluxDB.LineProtocol
+{
+    public class LineProtocolWriter
+    {
+        private static readonly DateTime UnixEpoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+
+        private readonly TextWriter textWriter;
+
+        private LinePosition position = LinePosition.NothingWritten;
+
+        public LineProtocolWriter()
```

Comment:
Perhaps adding a second constructor with a StringBuilder/StringWriter as parameter ? So we could manage the capacity allocation by ourselves in specific cases.